### PR TITLE
[EOC-277] Archived cohort view disable restore and delete for member

### PR DESCRIPTION
**Pull request by alemikolo**
I merged branch EOC-278 to this branch to avoid some errors which were fixed there.

### CODE REVIEW

File: client/src/common/components/ArchivedMessage.jsx [hunk 1]
```diff
@@ -59,25 +60,27 @@ class ArchivedMessage extends PureComponent {
...
-              className="archived-message__button primary-button"
-              disabled={pending}
-              type="button"
-              onClick={this.handleRestoring}
-            >
-              <FormattedMessage id="common.archived-message.restore-button" />
-            </button>
-            {pending && !isDialogVisible && <Preloader />}
-          </div>
+          {isOwner && (
```
Comment (adamklepacz) [inline client/src/common/components/ArchivedMessage.jsx]: you can also early return above this return 
```

if(!isOwner) {
    return null;
}
```

File: client/src/common/exceptions/NotFoundException.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export class NotFoundException extends Error {}
```
Comment (adamklepacz) [inline client/src/common/exceptions/NotFoundException.js]: Please also rename this to SourceNotFoundException

File: client/src/common/exceptions/index.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable import/default */
+import { AbortPromiseException } from './AbortPromiseException';
+import { NotFoundException } from './NotFoundException';
+import { UnauthorizedException } from './UnauthorizedException';
+import { ValidationException } from './ValidationException';
+
+export {
+  AbortPromiseException,
+  NotFoundException,
```
Comment (adamklepacz) [inline client/src/common/exceptions/index.js]: SoucreNotFoundException

File: client/src/modules/cohort/index.jsx [hunk 4]
```diff
@@ -81,17 +90,32 @@ class Cohort extends PureComponent {
...
-      })
-      .finally(() => {
         this.handleBreadcrumbs();
         this.setState({ pendingForDetails: false });
+      })
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pendingForDetails: false });
+        }
+        noOp();
```
Comment (adamklepacz) [inline client/src/modules/cohort/index.jsx]: @marekrozmus  is this noOp needed here? 
if an err occurs we just ignore them. Do we have to run noOp function here which takes time?

File: server/controllers/cohort.js [hunk 5]
```diff
@@ -506,7 +512,7 @@ const leaveCohort = (req, resp) => {
 
       const { memberIds, ownerIds } = doc;
 
-      if (isOwner(doc, userId)) {
+      if (isCohortOwner(doc, userId)) {
```
Comment (adamklepacz) [inline server/controllers/cohort.js]: in line 186. you have this assigned to a variable and here you are calling this within if. 
Please select one approach to be consistent

File: client/src/modules/cohort/index.jsx (hunk 4 above)
Comment (alemikolo) [reply client/src/modules/cohort/index.jsx]: I removed noOp

**Pull request merged**